### ISSUE TITLE
Normals display incorrectly in Blender, but correctly when exported

### ISSUE THREAD

**Issue by CFox04** [Skylumz/Sollumz#89]

**Comment by Skylumz** [comment]
best option is just make a import bool property "new normals" and if the user clicks it, it imports all models with calc normals split

**Comment by CFox04** [comment]
I agree

**Comment by CFox04** [comment]
There is more to this issue. The normals appear incorrect when exporting as well. Recreate by importing and exporting this:
[apa_ch2_04_house02.ydr.zip](https://github.com/Skylumz/Sollumz/files/7476701/apa_ch2_04_house02.ydr.zip) . What made you assume they appear correctly when exported?

**Comment by CFox04** [comment]
Also this model I attached has more vertices than the original in some places: 
![image](https://user-images.githubusercontent.com/46362482/140389189-bc861ed0-4963-44be-ae2e-db377811a7d0.png)

**Comment by CFox04** [comment]
[This commit](https://github.com/Skylumz/Sollumz/commit/2f953b9de7ece6f2e0eb1ac10689f93df1c97df4) caused the issue, which I'm sure you already knew. What I don't understand is why the normals still show up wrong when exported. It would makes sense that they appear wrong in Blender, but not when exported...

**Comment by CFox04** [comment]
So the issue is the order of these functions: 
```py
mesh.normals_split_custom_set_from_vertices(normals)
mesh.polygons.foreach_set("use_smooth", [True] * len(mesh.polygons))
````
should be: 
```py
mesh.polygons.foreach_set("use_smooth", [True] * len(mesh.polygons))
mesh.normals_split_custom_set_from_vertices(normals)
````

However, this causes more vertices to be exported. If you run the merge normals operator and export, then the number of vertices are correct (correct as in like they were before in the screenshot I posted). So we need to merge the normals on export.

**Comment by CFox04** [comment]
"However, this causes more vertices to be exported. If you run the merge normals operator and export, then the number of vertices are correct (correct as in like they were before in the screenshot I posted). So we need to merge the normals on export." I was wrong here. It only exported more verts because I commented out the `foreach_set` call. After adding it back in this issue is fixed! It was only the order of those functions.
